### PR TITLE
Add po revision list rest endpoint

### PR DESCRIPTION
**Pull request by leebradley**

### CODE REVIEW

File: sdk/src/rest_api/resources/purchase_order/v1/handler.rs [hunk 1]
```diff
@@ -93,3 +96,51 @@ pub fn get_purchase_order<'a>(
...
+            }
+            PurchaseOrderStoreError::ConstraintViolationError(err) => {
+                ErrorResponse::new(400, &format!("{}", err))
+            }
+            PurchaseOrderStoreError::ResourceTemporarilyUnavailableError(_) => {
+                ErrorResponse::new(503, "Service Unavailable")
+            }
+            PurchaseOrderStoreError::NotFoundError(_) => ErrorResponse::new(
+                404,
+                &format!("Purchase order {} not found", purchase_order_uid),
```
Comment (leebradley) [inline sdk/src/rest_api/resources/purchase_order/v1/handler.rs:126]: I need to update this error message to include the revision id

File: sdk/src/rest_api/resources/purchase_order/v1/payloads.rs [hunk 2]
```diff
@@ -64,8 +67,25 @@ pub struct PurchaseOrderVersionSlice {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct PurchaseOrderRevisionSlice {
-    revision_number: u64,
+    revision_number: String,
```
Comment (dnewh) [inline sdk/src/rest_api/resources/purchase_order/v1/payloads.rs:70]: Revision number was updated to be an i64 in #949

**Review by dnewh — APPROVED** [review]

File: sdk/src/rest_api/resources/purchase_order/v1/payloads.rs (hunk 2 above)
Comment (leebradley) [reply sdk/src/rest_api/resources/purchase_order/v1/payloads.rs]: That makes a lot of sense! Updated to use i64

File: sdk/src/rest_api/actix_web_3/routes/purchase_orders.rs [hunk 3]
```diff
@@ -152,15 +152,34 @@ pub async fn get_purchase_order_version(
 
 #[get("/purchase-order/{uuid}/version/{version_id}/revisions")]
```
Comment (shannynalayna) [inline sdk/src/rest_api/actix_web_3/routes/purchase_orders.rs]: s/versions/version & s/revisions/revision in the commit message

**Review by dnewh — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Review by dnewh — APPROVED** [review]

**Pull request merged**